persistence-cassandra/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/cassandra/CassandraReadSideHandler.scala

```diff
@@ -28,36 +29,44 @@ import scala.concurrent.{ ExecutionContext, Future }
  * Internal API
  */
 private[cassandra] abstract class CassandraReadSideHandler[Event <: AggregateEvent[Event], Handler](
-  session: CassandraSession, handlers: Map[Class[_ <: Event], Handler], dispatcher: String
+  session:    CassandraSession,
+  handlers:   Map[Class[_ <: Event], Handler],
+  dispatcher: String
 )(implicit ec: ExecutionContext) extends ReadSideHandler[Event] {
 
   private val log = LoggerFactory.getLogger(this.getClass)
 
   protected def invoke(handler: Handler, event: Event, offset: Offset): CompletionStage[JList[BoundStatement]]
 
+  protected def offsetStatement(offset: Offset): JList[BoundStatement]
+
   override def handle(): Flow[Pair[Event, Offset], Done, _] = {
-    akka.stream.scaladsl.Flow[Pair[Event, Offset]].mapAsync(parallelism = 1) { pair =>
-      handlers.get(pair.first.getClass.asInstanceOf[Class[Event]]) match {
-        case Some(handler) =>
-          for {
-            statements <- invoke(handler, pair.first, pair.second).toScala
-            done <- statements.size match {
-              case 0 => Future.successful(Done.getInstance())
-              case 1 => session.executeWrite(statements.get(0)).toScala
-              case _ =>
-                val batch = new BatchStatement
-                val iter = statements.iterator()
-                while (iter.hasNext)
-                  batch.add(iter.next)
-                session.executeWriteBatch(batch).toScala
-            }
-          } yield done
-        case None =>
-          if (log.isDebugEnabled)
-            log.debug("Unhandled event [{}]", pair.first.getClass.getName)
-          Future.successful(Done.getInstance())
+
+    def executeStatements(statements: JList[BoundStatement]): Future[Done] =
+      statements.size match {
+        case 0 => Future.successful(Done.getInstance())
+        case 1 => session.executeWrite(statements.get(0)).toScala
+        case _ =>
+          val batch = new BatchStatement
+          val iter = statements.iterator()
+          while (iter.hasNext)
+            batch.add(iter.next)
+          session.executeWriteBatch(batch).toScala
       }
-    }.withAttributes(ActorAttributes.dispatcher(dispatcher)).asJava
+
+    akka.stream.scaladsl.Flow[Pair[Event, Offset]]
+      .mapAsync(parallelism = 1) { pair =>
+
+        handlers.get(pair.first.getClass.asInstanceOf[Class[Event]]) match {
+          case Some(handler) =>
+            invoke(handler, pair.first, pair.second).toScala.flatMap(executeStatements)
+
+          case None =>
+            if (log.isDebugEnabled) log.debug("Unhandled event [{}]", pair.first.getClass.getName)
+            executeStatements(offsetStatement(pair.second))
+        }
+
```

Comment:
Instead of this pattern matching, I would have prefer to work with a `NoOps` handler emitting an empty list of `BoundStatement`, but I couldn't manage to build one. I'm get some compilation error (won't give the details here).

A `NoOps` handler is a much cleaner solution because it will increase local reasoning.  The following would be much simple implementation.

```
val handler = 
  handlers.getOrElse(
    pair.first.getClass.asInstanceOf[Class[Event]]),
    NoOpsHandler
  )
  
   invoke(handler, pair.first, pair.second).toScala.flatMap(executeStatements)
``` 

The `invoke` method would always add the offset statement and we avoid the branching and excessive logic.